src/borg/repository.py

```diff
@@ -308,8 +308,13 @@ def save_config(self, path, config):
                 # some python ports have no os.link, see #4901
                 logger.warning(link_error_msg)
 
-        with SaveFile(config_path) as fd:
-            config.write(fd)
+        try:
+            with SaveFile(config_path) as fd:
+                config.write(fd)
+        except PermissionError:
+            # error is only a problem if we even had a lock
+            if self.do_lock:
+                raise
```

Comment:
Also, what do you think about logging the silently failing PermissionError as a warning?

```
except PermissionError as e:
    # error is only a problem if we even had a lock
    if self.do_lock:
        raise
    logger.warning('%s This is expected when working on read-only repositories.' % e)
```

Reply:
There might be a warning during borg check anyway: "Failed to securely erase old repository config file (hardlinks not supported>). Old repokey data, if any, might persist on physical storage."

Reply:
logging a warning sounds good. is there a `.` at the end of the exception message so this looks correct?

Reply:
The filename in single quotes:

> [Errno 13] Permission denied: '/home/dummy/test/config.tmp' This is expected when working on read-only repositories.
> [Errno 13] Permission denied: '/home/dummy/test/nonce.tmp' This is expected when working on read-only repositories.

Maybe the `[Errno 13]` is confusing. Since PermissionError is an OSError, we could also assemble a custom message from `e.strerror` and `e.filename` https://docs.python.org/3/library/exceptions.html#OSError

Reply:
With https://github.com/borgbackup/borg/pull/5048/commits/69a985ab14f459140b7aad12e72f360e9b29d60f, output would look something like this:

>     $ borg check --verify-data --bypass-lock --debug test
>     using builtin fallback logging configuration
>     37 self tests completed in 0.10 seconds
>     Starting repository check
>     Verified integrity of /home/dummy/test/index.12
>     Read committed index of transaction 12
>     Segment transaction is    12
>     Determined transaction is 12
>     Found 13 segments
>     Failed to securely erase old repository config file (hardlinks not supported>). Old repokey data, if any, might persist on physical storage.
>     Permission denied: Failed writing to '/home/dummy/test/config.tmp'. This is expected when working on read-only repositories.
>     finished segment check at segment 12
>     Failed to securely erase old repository config file (hardlinks not supported>). Old repokey data, if any, might persist on physical storage.
>     Permission denied: Failed writing to '/home/dummy/test/config.tmp'. This is expected when working on read-only repositories.
>     Starting repository index check
>     Index object count match.
>     Finished full repository check, no problems found.
>     Starting archive consistency check...
>     Verified integrity of /home/dummy/test/index.12
>     Starting cryptographic data integrity verification...
>     Finished cryptographic data integrity verification, verified 8 chunks with 0 integrity errors.
>     TAM-verified manifest
>     Analyzing archive one (1/2)
>     Permission denied: Failed writing to '/home/dummy/test/nonce.tmp'. This is expected when working on read-only repositories.
>     Analyzing archive two (2/2)
>     Archive consistency check complete, no problems found.